publisher/internal/analytics/ethereum/types.go

```diff
@@ -34,3 +36,59 @@ type TokenTransaction struct {
 	Amount float64
 	Price  float64
 }
+
+type EventInstruction struct {
+	Name      string
+	Header    string
+	Signature string
+	PublishTo string
+	Operation
+}
+
+type WrappedEventLog struct {
+	Log          EventLog
+	Instructions EventInstruction
+}
+
+func (wel *WrappedEventLog) hasTopics() bool {
+	for _, str := range wel.Log.Topics {
+		if str != "" {
+			return true
+		}
+	}
+	return false
+}
+
+func (wel *WrappedEventLog) CreateInstructions() {
+	signature := wel.Log.Topics[0]
+	switch {
+	case strings.HasPrefix(signature, transferSig):
```

Comment:
Perhaps it makes sense to move these signature checks to `EventLog` structure as a method, e.g. `IsTransfer`. Similarly with other event typess.